mempool/v1/reactor.go

```diff
@@ -190,38 +249,185 @@ func (memR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
 
 	for {
 		// In case of both next.NextWaitChan() and peer.Quit() are variable at the same time
 		if !memR.IsRunning() || !peer.IsRunning() {
 			return
 		}
+		// This happens because the CElement we were looking at got garbage
+		// collected (removed). That is, .NextWait() returned nil. Go ahead and
+		// start from the beginning.
+		if next == nil {
+			select {
+			case <-memR.sidecar.TxsWaitChan(): // Wait until a tx is available in sidecar
+				// if a tx is available on sidecar, if fire is set too, then fire
+				if next = memR.sidecar.TxsFront(); next == nil {
+					continue
+				}
+			case <-peer.Quit():
+				return
+			case <-memR.Quit():
+				return
+			}
+		}
+		// Make sure the peer is up to date.
+		peerState, ok := peer.Get(types.PeerStateKey).(PeerState)
+		if !ok {
+			// Peer does not have a state yet. We set it in the consensus reactor, but
+			// when we add peer in Switch, the order we call reactors#AddPeer is
+			// different every time due to us using a map. Sometimes other reactors
+			// will be initialized before the consensus reactor. We should wait a few
+			// milliseconds and retry.
+			time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+			continue
+		}
+
+		if scTx, okConv := next.Value.(*mempool.SidecarTx); okConv && isSidecarPeer {
+			memR.Logger.Debug(
+				"broadcasting a sidecarTx to peer",
+				"peer", peerID,
+				"tx", scTx.Tx.Hash(),
+			)
+			if peerState.GetHeight() < scTx.DesiredHeight-1 {
+				time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+				continue
+			}
+			if _, ok := scTx.Senders.Load(peerID); !ok {
+				msg := &protomem.MEVTxs{
+					Txs:           [][]byte{scTx.Tx},
+					DesiredHeight: scTx.DesiredHeight,
+					BundleId:      scTx.BundleID,
+					BundleOrder:   scTx.BundleOrder,
+					BundleSize:    scTx.BundleSize,
+					GasWanted:     scTx.GasWanted,
+				}
+				success := p2p.SendEnvelopeShim(peer, p2p.Envelope{ //nolint: staticcheck
+					ChannelID: mempool.SidecarChannel,
+					Message:   msg,
+				}, memR.Logger)
+				// if sending over envelop protocol fails, try over legacy protocol
+				if !success {
+					msg := &protomem.MEVMessage{
+						Sum: &protomem.MEVMessage_Txs{
+							Txs: &protomem.Txs{Txs: [][]byte{scTx.Tx}},
+						},
+						DesiredHeight: scTx.DesiredHeight,
+						BundleId:      scTx.BundleID,
+						BundleOrder:   scTx.BundleOrder,
+						BundleSize:    scTx.BundleSize,
+						GasWanted:     scTx.GasWanted,
+					}
+
+					success := p2p.SendEnvelopeShim(peer, p2p.Envelope{
+						ChannelID: mempool.SidecarLegacyChannel,
+						Message:   msg,
+					}, memR.Logger)
+					if !success {
+						time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+						continue
+					}
+				}
+			} else {
+				memR.Logger.Info(
+					"broadcasting sidecarTx to peer failed",
+					"peer", peerID,
+					"was considered sidecarPeer", isSidecarPeer,
+					"was converted to sidecarTx", okConv,
+					"tx", scTx.Tx.Hash(),
+				)
+			}
```

Comment:
Same as in v0/reactor.go commented up:
This log message could be misleading - there are no issues with broadcasting sidecarTx  but the broadcasting is simply optimized and the peer is skipped if being the one that sent the transactions to the node.

Remove or modify the message, if it is useful for any reason.

Reply:
Listed in https://github.com/informalsystems/audit-skip/issues/6

Reply:
Removed in https://github.com/skip-mev/mev-tendermint/pull/131